source/server/listener_impl.h

```diff
@@ -461,9 +466,11 @@ class ListenerImpl final : public Network::ListenerConfig,
   void checkIpv4CompatAddress(const Network::Address::InstanceConstSharedPtr& address,
                               const envoy::config::core::v3::Address& proto_address);
 
-  void addListenSocketOptions(const Network::Socket::OptionsSharedPtr& options) {
-    ensureSocketOptions();
-    Network::Socket::appendOptions(listen_socket_options_, options);
+  void addListenSocketOptions(uint32_t address_index,
+                              const Network::Socket::OptionsSharedPtr& options) {
```

Comment:
nit: I think you can avoid passing an index here and actually just pass an already initialized options structure that you drive from the loop inside `buildListenSocketOptions`. It would be nice to not have to reason about the indexes in multiple places.

Reply:
If I move line 472 `ensureSocketOptions(address_index);` into the `buildListenSocketOptions`, then this method seems useless. In the end. In the end, it looks like I spread this two lines code into the `buildListenSocketOptions`